app/main.py

```diff
@@ -1,6 +1,19 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
+    cache = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Callable:
+        if args in cache:
+            print("Getting from cache")
+            return cache[args]
+        else:
+            print("Calculating new result")
+            cache[args] = func(*args)
```

Comment:
Code Efficiency: You are calling `func(*args)` and storing the result in `cache[args]`. Instead of calling `cache[args]` again in the return statement, you can directly return the stored result. This avoids accessing the dictionary twice unnecessarily.